specifyweb/frontend/js_src/lib/components/QueryBuilder/Export.tsx

```diff
@@ -87,18 +127,27 @@ export function QueryExportButtons({
         <QueryButton
           disabled={fields.length === 0}
           showConfirmation={showConfirmation}
-          onClick={(): void =>
-            doQueryExport(
-              '/stored_query/exportcsv/',
-              userPreferences.get(
-                'queryBuilder',
-                'behavior',
-                'exportFileDelimiter'
-              )
-            )
-          }
+          onClick={(): void => {
+            selectedRows.size === 0
+              ? doQueryExport(
+                  '/stored_query/exportcsv/',
+                  userPreferences.get(
+                    'queryBuilder',
+                    'behavior',
+                    'exportFileDelimiter'
+                  )
+                )
+              : downloadFile(
+                  `${
+                    queryResource.isNew()
+                      ? 'new query export'
```

Comment:
instead of `new query export`, which is not localized, and not informative, can you do something like this:

```ts
`${queryText.newQueryName()} - ${schema.models[baseTableName].label}`
```